smarts/env/gymnasium/wrappers/metric/costs.py

```diff
@@ -559,30 +628,55 @@ def get_dist(road_map: RoadMap, point_a: Point, point_b: Point) -> float:
         ),
     )
     plan = Plan(road_map=road_map, mission=mission, find_route=False)
-    plan.create_route(mission=mission, radius=20)
+    plan.create_route(mission=mission, start_lane_radius=3, end_lane_radius=0.5)
     assert isinstance(plan.route, RoadMap.Route)
     from_route_point = RoadMap.Route.RoutePoint(pt=point_a)
     to_route_point = RoadMap.Route.RoutePoint(pt=point_b)
 
     dist_tot = plan.route.distance_between(start=from_route_point, end=to_route_point)
     if dist_tot == None:
         raise CostError("Unable to find road on route near given points.")
-    elif dist_tot < 0:
-        # This happens when agent overshoots the goal position while
-        # remaining outside the goal capture radius at all times. Default
-        # positional goal radius is 2m.
-        dist_tot = abs(dist_tot)
-
-    # Account for agent ending in a different lane but in the same road as
-    # the goal position.
-    start_lane = plan.route.start_lane
-    end_lane = plan.route.end_lane
-    lane_error = abs(start_lane.index - end_lane.index)
-    if len(plan.route.roads) == 1 and lane_error > 0:
-        assert start_lane.road == end_lane.road
-        end_offset = end_lane.offset_along_lane(world_point=point_b)
-        lane_width, _ = end_lane.width_at_offset(end_offset)
-        lane_error_dist = lane_error * lane_width
-        dist_tot += lane_error_dist
-
-    return dist_tot
+    elif dist_tot < 0 and not tolerate:
+        raise CostError(
+            "Route computed in reverse direction from point_b to "
+            f"point_a resulting in negative distance: {dist_tot}."
```

Comment:
Perhaps we should be using the term `displacement` rather than distance here since `distance` can also mean magnitude from a point.

Reply:
If the route from `point_b` to `point_a` is not straight, then the route `distance` will be longer than the `displacement` between `point_b` and `point_a`. So rephrasing the error message as "Route computed in reverse direction from point_b to point_a resulting in negative displacement: {dist_tot}." also sounds a little awkward as the displacement might be shorter than the printed `dist_tot`.